File: ts-template/src/handler.ts
type FnContext = {
  method?: string;
  path?: string;
  query?: Record<string, string>;
  headers?: Record<string, unknown>;
  cloudevent?: any;
  log?: { info: (...a: any[]) => void; error: (...a: any[]) => void };
  cloudEventResponse?: (data: any) => any;
};

type FnInstance = {
  init?: () => void | Promise<void>;
  shutdown?: () => void | Promise<void>;
  handle: (ctx: FnContext, data: any) => any;
};

function newFn(): FnInstance {
  return {
    handle: async (ctx, _data) => {
      const m = ctx.method || "GET";
      const p = ctx.path || "/";

      ctx.log?.info?.({ msg: "fn_handle", method: m, path: p });

      if (m === "GET" && p === "/") {
        return { statusCode: 200, body: "Hello from STACKIT Functions" };
      }

      return {
        statusCode: 404,
        body: { error: "not_found", method: m, path: p },
      };
    },
  };
}

// Contract: wrapper expects either export object with .new() or instance
export = { new: newFn };

File: rs-template/src/handler.rs
use crate::config::HandlerConfig;
use actix_web::{HttpRequest, HttpResponse, http::Method, web::Data};
use log::info;

// Implement your function's logic here
pub async fn index(req: HttpRequest, config: Data<HandlerConfig>) -> HttpResponse {
    info!("{:#?}", req);
    if req.method() == Method::GET && req.path() == "/" {
       return HttpResponse::Ok().body("Hello from STACKIT Functions");
    }

    HttpResponse::NotFound().body("not_found")
}
